### PR TITLE
[Ast] Implement ascii, chr function in ast

### PR DESCRIPTION
**Pull request by seonghun-dev**
Resolve : https://github.com/gluesql/gluesql/issues/798

### CODE REVIEW

File: core/src/executor/evaluate/function.rs [hunk 1]
```diff
@@ -223,6 +223,35 @@ pub fn substr(
     Ok(Value::Str(string))
 }
 
+pub fn ascii(name: String, expr: Evaluated<'_>) -> Result<Value> {
+    let string = eval_to_str!(name, expr);
+
+    if string.len() != 1 {
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:229]:
```suggestion
    if !string.is_ascii() {
```

File: core/src/executor/evaluate/function.rs [hunk 2]
```diff
@@ -223,6 +223,35 @@ pub fn substr(
...
+        return Err(EvaluateError::AsciiFunctionRequiresSingleCharacterValue.into());
+    }
+
+    let char = string.chars().next();
+    match char {
+        Some(char) => {
+            let char = char as u8;
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:236]:
```suggestion
```

duplicated type casting

File: core/src/executor/evaluate/function.rs [hunk 3]
```diff
@@ -223,6 +223,35 @@ pub fn substr(
...
+pub fn chr(name: String, expr: Evaluated<'_>) -> Result<Value> {
+    let expr = eval_to_int!(name, expr);
+
+    match expr {
+        0..=255 => {
+            let expr = expr as u8;
+            Ok(Value::Str((expr as char).to_string()))
+        }
+        _ => Err(EvaluateError::ChrFunctionRequiresIntegerValueInRange0To255.into()),
+    }
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:252]: It seems good to convert to `String` here too and check with `is_ascii`. Then it seems that no conversion is necessary for the return type `char` .

File: core/src/executor/evaluate/function.rs [hunk 4]
```diff
@@ -223,6 +223,35 @@ pub fn substr(
...
 }
 
+pub fn ascii(name: String, expr: Evaluated<'_>) -> Result<Value> {
+    let string = eval_to_str!(name, expr);
+
+    if string.len() != 1 {
+        return Err(EvaluateError::AsciiFunctionRequiresSingleCharacterValue.into());
+    }
+
+    let char = string.chars().next();
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:233]: `char` is the name of a primitive type, so we recommend something else.
ex) `character`

File: core/src/executor/evaluate/function.rs (hunk 1 above)
Comment (ever0de) [reply core/src/executor/evaluate/function.rs:229]: edit) 

```rust
    let string = eval_to_str!(name, expr);
    let mut iter = string.chars();

    match iter.next() {
        Some(text) => {
            if iter.count() > 0 {
                // non char
            }

            if text.is_ascii() {
                // return Ok(Value/...)
            } else {
                // non ascii string
            }

            todo!()
        }
        // empty string?
        None => todo!(),
    }
```

File: core/src/executor/evaluate/function.rs (hunk 3 above)
Comment (ever0de) [reply core/src/executor/evaluate/function.rs:252]: This is a bad comment because the `CHR` function only accepts u8(numeric).

**Review by ever0de — APPROVED** [review]

**Review by panarch — APPROVED** [review]
All looks nice! Thanks a lot 👍

**Pull request merged**